backend/routes/songs.js

```diff
@@ -120,6 +120,59 @@ app.put("/likes/:id", (req, res, next) => {
     });
 });
 
+
+
+
+
+app.get("/search", (req, res, next) => {
+    console.log(`user seesion: ${req.session.userId}`);
+    const pageSize = +req.query.pageSize;
+    const currPage = +req.query.page;
+    const songName = req.query.songName;
+    const artistName = req.query.artistName;
+    const genreName = req.query.genreName;
+
+    let fetchedSongs;
+    let query = {};
+
+    if(songName!=='') query["name"] = songName;
+    if(artistName!=='') query["artists.display_name"] = artistName;
+    if(genreName!=='') query["genre"] = genreName;
+
+
+    const songQuery = Song.find(query);
+
+    Song.find(query).then(results=>{
```

Comment:
Why twice?

Reply:
to check how mush in total, the other one is maximum 10